### DISCUSSION TITLE
About SpikeRate loss and final classification

### DISCUSSION THREAD

**Discussion by GBrayshaw** [lava-nc/lava-dl#72] [Q&A] [answered]
Hi,

I'm curious as to what effect the parameters for slayer.loss.SpikeRate (true_rate & false_rate) have on the output spike train from a network. For example:

true_rate = 0.5
false_rate = 0
sample time = 1000ms
sample length = 1ms
classifier=slayer.classifier.Rate.predict

Given the parameters above, should the correct output neuron look to spike 500 times over the course of the 1000ms (0.5 of the time)?

Similarly, is the final classification from the network given as the output neuron with the spike rate closest to 500 spikes or simply the neuron with the highest spike rate output during the sample time?

Thank you in advance for your help.

**Comment by bamsumit** [comment, marked as answer]
> Given the parameters above, should the correct output neuron look to spike 500 times over the course of the 1000ms (0.5 of the time)?

Hi @Geoforger, assuming you are not using a moving window, yes, the `SpikeRate` loss drives the network output to spike with an overall rate of 0.5 i.e. 500 spikes in 1000 timesteps when the neuron is true and 0 times when the neuron is false.

> Similarly, is the final classification from the network given as the output neuron with the spike rate closest to 500 spikes or simply the neuron with the highest spike rate output during the sample time?

`slayer.classifier.Rate.predict` predict the neuron with the highest spiking rate as the class. Documentation [here](https://lava-nc.org/lava-lib-dl/slayer/classifier.html#lava.lib.dl.slayer.classifier.Rate.predict). It does not take into account the loss target.
